--- usr/local/lib/mbedos/mbed-os/tools/toolchains/gcc.py.orig	2020-06-20 11:44:48.097729530 -0700
+++ usr/local/lib/mbedos/mbed-os/tools/toolchains/gcc.py	2020-06-20 11:45:25.790231616 -0700
@@ -122,13 +122,13 @@
         # FPU handling, M7 possibly to have double FPU
         if core == "Cortex-M4F":
             self.cpu.append("-mfpu=fpv4-sp-d16")
-            self.cpu.append("-mfloat-abi=softfp")
+            self.cpu.append("-mfloat-abi=hard")
         elif core == "Cortex-M7F" or core.startswith("Cortex-M33F"):
             self.cpu.append("-mfpu=fpv5-sp-d16")
-            self.cpu.append("-mfloat-abi=softfp")
+            self.cpu.append("-mfloat-abi=hard")
         elif core == "Cortex-M7FD":
             self.cpu.append("-mfpu=fpv5-d16")
-            self.cpu.append("-mfloat-abi=softfp")
+            self.cpu.append("-mfloat-abi=hard")
 
         if target.core == "Cortex-A9":
             self.cpu.append("-mthumb-interwork")
